napari/layers/utils/_tests/test_text_manager.py

```diff
@@ -544,3 +551,161 @@ def test_view_text_with_constant_text():
     # because assert_array_equal broadcasts scalars automatically
     assert len(actual) == 2
     np.testing.assert_array_equal(actual, ['A', 'A'])
+
+
+def test_init_with_constant_color():
+    color = {'constant': 'red'}
+    features = pd.DataFrame(index=range(3))
+
+    text_manager = TextManager(color=color, features=features)
+
+    actual = text_manager.color._values
+    assert_colors_equal(actual, 'red')
+
+
+def test_init_with_manual_color():
+    color = ['red', 'green', 'blue']
+    features = pd.DataFrame({'class': ['A', 'B', 'C']})
+
+    text_manager = TextManager(color=color, features=features)
+
+    actual = text_manager.color._values
+    assert_colors_equal(actual, ['red', 'green', 'blue'])
+
+
+def test_init_with_derived_color():
+    color = {'feature': 'colors'}
+    features = pd.DataFrame({'colors': ['red', 'green', 'blue']})
+
+    text_manager = TextManager(color=color, features=features)
+
+    actual = text_manager.color._values
+    assert_colors_equal(actual, ['red', 'green', 'blue'])
+
+
+def test_init_with_derived_color_missing_feature():
+    color = {'feature': 'not_a_feature'}
+    features = pd.DataFrame({'colors': ['red', 'green', 'blue']})
+
+    with pytest.warns(RuntimeWarning):
+        text_manager = TextManager(color=color, features=features)
+
+    actual = text_manager.color._values
+    assert_colors_equal(actual, ['cyan'] * 3)
+
+
+def test_apply_with_constant_color():
+    color = {'constant': 'red'}
+    features = pd.DataFrame({'class': ['A', 'B', 'C']})
+    text_manager = TextManager(color=color, features=features)
+
+    features = pd.DataFrame({'class': ['A', 'B', 'C', 'D', 'E']})
+    text_manager.apply(features)
+
+    actual = text_manager.color._values
+    assert_colors_equal(actual, 'red')
+
+
+def test_apply_with_manual_color():
```

Comment:
nit: pls add comments to tests like these to specify expected behaviour e.g. "first three features should be red, green, and blue as specified, while the remaining two default to yellow"

Reply:
I usually interpret explanatory/implementation comments in tests as a bit of a code smell, so I try to avoid them unless they're really needed to explain complex or confusing behavior (which is sometimes hard to avoid). But I do like including expected behavior in test names themselves, because that's something you see in test outputs, so it makes it clearer what behavior broke when this test fails. So I did that here.

Otherwise, I think the test is pretty readable as it is and the comment above doesn't add a huge amount in terms of explanatory power after updating the test name. Obviously, I have a pretty biased perspective as the author of the test and implementation, so definitely happy to add a comment if really desired, but as this is a nit I'll leave it for now.